lib/common/optional/types.go

```diff
@@ -364,3 +364,72 @@ func (x *Body) Value() []byte {
 func (x *Body) Exists() bool {
 	return x.exists
 }
+
+// Result represents a Result type.
```

Comment:
could you move this to a different file / sub-package, since `Result` isn't an `Optional`